src/wrappers/themis/wasm/src/secure_message.js

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) 2019 Cossack Labs Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ * Themis Secure Message.
+ */
+
+const libthemis = require('./libthemis.js')
+const keygen = require('./secure_keygen.js')
+const errors = require('./themis_error.js')
+const utils = require('./utils.js')
+
+const subsystem = 'SecureMessage'
+
+const ThemisError = errors.ThemisError
+const ThemisErrorCode = errors.ThemisErrorCode
+
+class SecureMessage {
+    constructor(keyPair) {
+        if (arguments.length == 1) {
+            if (!(keyPair instanceof keygen.KeyPair)) {
+                throw new ThemisError(subsystem, ThemisErrorCode.INVALID_PARAMETER,
+                    'invalid argument: must be KeyPair')
+            }
+            this.privateKey = keyPair.privateKey
+            this.publicKey = keyPair.publicKey
+            return
+        }
+
+        if (arguments.length == 2) {
+            let arg0isPrivateKey = arguments[0] instanceof keygen.PrivateKey
+            let arg0isPublicKey  = arguments[0] instanceof keygen.PublicKey
+            let arg1isPrivateKey = arguments[1] instanceof keygen.PrivateKey
+            let arg1isPublicKey  = arguments[1] instanceof keygen.PublicKey
+
+            if (arg0isPublicKey && arg1isPrivateKey) {
+                this.publicKey = arguments[0]
+                this.privateKey = arguments[1]
+                return
+            }
+            if (arg0isPrivateKey && arg1isPublicKey) {
```

Comment:
it's so cute) you allow users to use any order of params))

Reply:
that's really cool, i hope it will improve usability, not add confusion :D